### ISSUE TITLE
Poly-encoder validation results args for convai2 

### ISSUE THREAD

**Issue by bhattaca** [facebookresearch/ParlAI#2695]
Hi, 
qq, what are the right parameters to get validation R@20  numbers for poly-encoder fine-tuned for convai2 task.

Is this enough : 
examples/eval_model.py -mf ~/ParlAIModels/bert_polyencoder_fine_tuned_1 -t convai2 --eval-candidates inline -dt valid 

specially for candidates related args.

**Comment by stephenroller** [comment]
Yeah that looks right to me. `python examples/eval_model.py -t convai2 -mf 'zoo:pretrained_transformers/model_poly/model' -bs 32 -ecands inline` should evaluate the pretrained model that we released. I just ran it and got 89.37 which looks like the 89.5+/-0.1 in table 10. I'll let @klshuster confirm tho.

(PS kurt it occurs to me that there isn't a test for this)

**Comment by klshuster** [comment]
yeah that should be the correct params for evaluation on convai2. and good point @stephenroller, we should add one (i'll make a separate issue)

I'll close this issue for now, please let us know if you have any other issues @bhattaca!

**Comment by bhattaca** [comment]
'hits@20'  does not work with --metrics. or am I missing something obvious.

**Comment by stephenroller** [comment]
No, you’d have to edit metrics.py to add 20 to the list of values we compute hits@ for. Currently we do 1, 10, 100

**Comment by klshuster** [comment]
are you curious about hits@20 or hits@1/20 - that is, are you interested in measuring the accuracy out of 20 candidates, or the recall in the first 20 candidates? If the former (which is what we reported in the paper), then you're interested in the `hits@1` metric, as the convai2 validation set with `inline` candidates provides 20 candidates to the model by default

**Comment by bhattaca** [comment]
@klshuster Thanks, thats exactly what I was looking for. Looks like the higher accuracy ~89 is heavily dependent on the batchsize, I am unfortunately restricted by hardware. 
the 'first_m' codes (m>=64) is very competitive but i am unable to get past 86~ accuracy without higher batch size.

**Comment by klshuster** [comment]
indeed, we found that higher batchsize helps the model generalize as it is trained to rank the correct candidate amongst a greater number of negatives
